core/src/main/java/io/grpc/internal/RetriableStream.java

```diff
@@ -942,8 +935,27 @@ private RetryPlan makeRetryDecision(Status status, Metadata trailer) {
         } // else no retry
       } // else no retry
 
-      return new RetryPlan(
-          shouldRetry, /* isFatal = */ false, backoffNanos, isHedging ? pushbackMillis : null);
+      return new RetryPlan(shouldRetry, backoffNanos);
+    }
+
+    private HedgingPlan makeHedgingDecision(Status status, Metadata trailer) {
+      String pushbackStr = trailer.get(GRPC_RETRY_PUSHBACK_MS);
+      Integer pushbackMillis = null;
+      if (pushbackStr != null) {
+        try {
+          pushbackMillis = Integer.valueOf(pushbackStr);
+        } catch (NumberFormatException e) {
+          pushbackMillis = -1;
+        }
+      }
```

Comment:
This looks identical to lines 906-914 in `makeRetryDecision`. Can it be extracted out?

Reply:
Done.